module/zfs/arc.c

```diff
@@ -4067,13 +4147,37 @@ arc_evict_state(arc_state_t *state, arc_buf_contents_t type, uint64_t spa,
 		multilist_sublist_unlock(mls);
 	}
 
+	if (use_evcttq) {
+		evarg = kmem_alloc(sizeof (*evarg) * nthreads, KM_NOSLEEP);
+		/*
+		 * Fall back to the regular single evict if it is not possible
+		 * to allocate memory for the task queue entries.
+		 */
+		if (evarg == NULL)
+			use_evcttq = B_FALSE;
+	}
+
 	/*
 	 * While we haven't hit our target number of bytes to evict, or
 	 * we're evicting all available buffers.
 	 */
 	while (total_evicted < bytes) {
 		int sublist_idx = multilist_get_random_index(ml);
 		uint64_t scan_evicted = 0;
+		uint64_t left = (bytes == ARC_EVICT_ALL ? bytes :
+		    bytes - total_evicted);
+
+		/*
+		 * Compute:
+		 * 1. Number of chunks (each MIN_EVICT_SIZE) to evict (nchunks)
+		 * 2. Number of tasks to run (ntasks)
+		 * 3. Bytes per task to evict (evict_pertask)
+		 * 4. Do some tasks take extra work? (extra_chunks)
+		 */
+		uint64_t nchunks = ((left - 1) / MIN_EVICT_SIZE) + 1;
+		unsigned ntasks = nchunks < nthreads ? nchunks : nthreads;
+		uint64_t evict_pertask = MIN_EVICT_SIZE * (nchunks / ntasks);
+		uint64_t extra_chunks = nchunks % ntasks;
```

Comment:
Again you are over-complicating something about MIN_EVICT_SIZE.  Why not just 
```suggestion
		uint64_t evict_pertask = DIV_ROUND_UP(left, ntasks);
```
and no `extra_chunks`?